src/featurizedatoms.jl

```diff
@@ -0,0 +1,50 @@
+using ..ChemistryFeaturization.AbstractType: AbstractAtoms, AbstractFeaturization
+
+"""
+    FeaturizedAtoms
+
+Container object for an `Atoms` object, a `featurization`, and the resulting
+`encoded_features` from applying the `featurization` to the `atoms`.
+
+Note: `encoded_features` will NOT change for a given atoms-featurization pair.
+"""
+struct FeaturizedAtoms{A<:AbstractAtoms,F<:AbstractFeaturization}
+    atoms::A
+    featurization::F
+    encoded_features::Any
+    FeaturizedAtoms{A,F}(
```

Comment:
Why is this constructor internal?

Reply:
Because we want to enforce that the `encoded_features` are actually produced by the `featurization` encoding on the `atoms`. If it were external (assuming I'm understanding correctly), then there would still be a constructor that allows those three to all be fed in independently...